deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/Parser.java

```diff
@@ -1,8 +1,35 @@
 package com.airbnb.deeplinkdispatch;
 
+import java.util.List;
+
 /**
  * Parser classes are used for keeping a registry of deep links and dispatching them from Intents.
  */
-public interface Parser {
-  DeepLinkEntry parseUri(String uri);
+public abstract class Parser {
+
+  public Parser(List<DeepLinkEntry> registry) {
+    this.registry = registry;
+  }
+
+  /**
+   * The List of DeepLinkEntries collected from a Module. Annotation Processing-generated Children
+   * of this class will pass the collected {@code List<DeepLinkEntry>} in the constructor's
+   * {@code super()}.
+   */
+  private final List<DeepLinkEntry> registry;
+
+  public List<DeepLinkEntry> getRegistry() {
```

Comment:
registry is a bit of an unusual name for a list. it isn't clear based on this class why that name is used

Reply:
Are you thinking of something more explicit than the comment at the top of this file? https://github.com/airbnb/DeepLinkDispatch/pull/260/files/256268ae06907ec3dcb220ace44280a99cc7e624#diff-54c0eadfe4fe7515a50210ca4ed3ceecR6

Reply:
A clearer name would be something like `getRegisteredDeeplinks`. you just generally don't see a generic collection referred to as a specific name like this